src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1104 @@
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+
+#include "service-area-server.h"
+#include "service-area-delegate.h"
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Commands.h>
+#include <app/AttributeAccessInterfaceRegistry.h>
+#include <app/CommandHandler.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/EventLogging.h>
+#include <app/InteractionModelEngine.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/util.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+using Status = chip::Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ServiceArea {
+
+// ****************************************************************************
+// Service Area Server Instance
+
+Instance::Instance(Delegate * aDelegate, EndpointId aEndpointId, BitMask<Feature> aFeature) :
+    AttributeAccessInterface(MakeOptional(aEndpointId), Id), CommandHandlerInterface(MakeOptional(aEndpointId), Id),
+    mDelegate(aDelegate), mEndpointId(aEndpointId), mClusterId(Id), mFeature(aFeature)
+{
+    ChipLogProgress(Zcl, "Service Area: Instance constructor");
+    mDelegate->SetInstance(this);
+}
+
+Instance::~Instance()
+{
+    InteractionModelEngine::GetInstance()->UnregisterCommandHandler(this);
+    unregisterAttributeAccessOverride(this);
+}
+
+CHIP_ERROR Instance::Init()
+{
+    ChipLogProgress(Zcl, "Service Area: INIT");
+
+    // Check if the cluster has been selected in zap
+    VerifyOrReturnError(emberAfContainsServer(mEndpointId, Id), CHIP_ERROR_INVALID_ARGUMENT,
+                        ChipLogError(Zcl, "Service Area: The cluster with Id %lu was not enabled in zap.", long(Id)));
+
+    ReturnErrorOnFailure(InteractionModelEngine::GetInstance()->RegisterCommandHandler(this));
+
+    VerifyOrReturnError(registerAttributeAccessOverride(this), CHIP_ERROR_INCORRECT_STATE);
+
+    return mDelegate->Init();
+}
+
+//*************************************************************************
+// core functions
+
+CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    ChipLogDetail(Zcl, "Service Area: Reading attribute");
+
+    switch (aPath.mAttributeId)
+    {
+
+    case Attributes::SupportedLocations::Id:
+        return ReadSupportedLocations(aEncoder);
+
+    case Attributes::SupportedMaps::Id:
+        return ReadSupportedMaps(aEncoder);
+
+    case Attributes::SelectedLocations::Id:
+        return ReadSelectedLocations(aEncoder);
+
+    case Attributes::CurrentLocation::Id:
+        return aEncoder.Encode(GetCurrentLocation());
+
+    case Attributes::EstimatedEndTime::Id:
```

Comment:
Should this attribute be marked as attribute-access-interface-only in `src/app/zap-templates/zcl/zcl.json` and `src/app/zap-templates/zcl/zcl-with-test-extensions.json`?

Same for the other non-list attributes here.

Reply:
I have added the non-list attributes to the `attributeAccessInterfaceAttributes` [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/555fac8a5bfa890ee4c47a9376b719ced97af0bf). Why don't we need to also add the list attributes? Are there assumed to be accessed via the attribute access interface by default?

Reply:
List attributes are only implementable via attribute access interface, yes, and ZAP knows that, so they get handled automatically.